src/src/PluginStructs/P105_data_struct.h

```diff
@@ -1,99 +1,81 @@
 #ifndef PLUGINSTRUCTS_P105_DATA_STRUCT_H
 #define PLUGINSTRUCTS_P105_DATA_STRUCT_H
 
+#include "../../_Plugin_Helper.h"
 #ifdef USES_P105
 
-# include "../../_Plugin_Helper.h"
-
-// https://wiki.liutyi.info/display/ARDUINO/AHT10
-// https://github.com/enjoyneering/AHT10/blob/master/src/AHT10.h
-// https://github.com/adafruit/Adafruit_AHTX0
-// https://github.com/arendst/Tasmota/blob/0650744ac27108931a070918f08173d71ddfd68d/tasmota/xsns_63_aht1x.ino
-
-# define AHT10_ADDRESS_0X38         0x38  // chip I2C address no.1 for AHT10/AHT15/AHT20, address pin connected to GND
-# define AHT10_ADDRESS_0X39         0x39  // chip I2C address no.2 for AHT10 only, address pin connected to Vcc
-
-# define AHT10_INIT_CMD             0xE1  // initialization command for AHT10/AHT15
-# define AHT20_INIT_CMD             0xBE  // initialization command for AHT20
-# define AHT10_START_MEASURMENT_CMD 0xAC  // start measurment command
-# define AHT10_NORMAL_CMD           0xA8  // normal cycle mode command, no info in datasheet!!!
-# define AHT10_SOFT_RESET_CMD       0xBA  // soft reset command
-
-# define AHT10_INIT_NORMAL_MODE     0x00  // enable normal mode
-# define AHT10_INIT_CYCLE_MODE      0x20  // enable cycle mode
-# define AHT10_INIT_CMD_MODE        0x40  // enable command mode
-# define AHT10_INIT_CAL_ENABLE      0x08  // load factory calibration coeff
-
-
-# define AHT10_DATA_MEASURMENT_CMD  0x33  // no info in datasheet!!! my guess it is DAC resolution, saw someone send 0x00 instead
-# define AHT10_DATA_NOP             0x00  // no info in datasheet!!!
+enum AHTx_device_type {
+  AHT10_DEVICE = 10,
+  AHT20_DEVICE = 20,
+  AHT21_DEVICE = 21,
+};
 
+enum AHTx_state {
```

Comment:
Can you make this `enum class` ? This does add the requirement to prefix the enum values with `AHTx_state::` and may require some more `static_cast` calls, but then the compiler can help detect if you missed handling an element in a switch statement and also gives a compiler error if you provide the wrong type as a function argument.
Otherwise it is just an `int` and you can easily mix up enum values order in a function call.